### PR TITLE
[EMBER]: upgrade to 2.8 and remove DS check

### PR DESCRIPTION
**Pull request by snewcomer**
References #214 

- [x] remove ember data check.  If user is not using ember-data, this check will fail?
- [x] Update tests to use `moduleForAcceptance`
- [x] Add docs for a few methods
- [x] remove ember-version-is
- [x] add test for custom store

### CODE REVIEW

**Comment by snewcomer** [comment]
Running this test [here](https://github.com/hhff/ember-infinity/blob/master/tests/acceptance/infinity-route-offset-trigger-test.js#L79) and seems as though in test land Ember's event handling has changed and for some reason the height/distance of items in the DOM are also different.  Need to figure out where/why this has changed to be able to modify the tests.

**master ember 2.4**
screen shot 2017-04-15 at 3 15 58 pm

**this branch ember 2.8**
screen shot 2017-04-15 at 3 19 42 pm

**Comment by snewcomer** [comment]
Ok so what is interesting is upgrading from `ember-cli-qunit` 1.4 to >2.1 is ok.  2.2 is what causes some scroll tests to fail.  Now can dig into that.

**Comment by snewcomer** [comment]
@hhff this PR is passing now.  A few other things other than upgrading are listed above.

File: ember-cli-build.js [hunk 1]
```diff
@@ -4,7 +4,9 @@ var EmberAddon = require('ember-cli/lib/broccoli/ember-addon');
 
 module.exports = function(defaults) {
   var app = new EmberAddon(defaults, {
-    // Add options here
+    babel: {
```
Comment (snewcomer) [inline ember-cli-build.js]: Take care of Object assign not existing in < ie 11

**Comment by hhff** [comment]
@snewcomer - in the past I've done `const assign = Object.assign || Ember.assign`.  Is that simpler?

**Comment by hhff** [comment]
@snewcomer - right now, a user can't use Ember Infinity without Ember Data (this.store.find or this.store.query are both ED methods)

**Comment by hhff** [comment]
re: Ember Data - I would like to remove the dep on Ember-Version-Is however...

**Comment by snewcomer** [comment]
@hhff I think the case for the util is if <=ie11 and no Ember.assign.  Also here is an acceptance test using a custom store and custom find method :).  Let me know your thoughts.

https://github.com/hhff/ember-infinity/pull/215/commits/450a142c71263e051c8ebbffade42ac53bb1d85b

**Comment by hhff** [comment]
Awesome work.  I'm cool with all of this.  Ok for me to merge?

**Comment by snewcomer** [comment]
@hhff yep if you are good with these changes!

**Pull request merged**